core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -439,11 +447,102 @@ public static Object getDataDataTypeForNoDictionaryColumn(String dimensionValue,
     }
   }
 
+  private static long createTimeInstant(String dimensionValue, String dateFormat) {
```

Comment:
this method is very long, can you add a detailed comments in method explaining what is the logic in this method with an example, so that it will be easy for review and for any other developer in future to change for any issues or improvements